app/bundles/UserBundle/Controller/ProfileController.php

```diff
@@ -237,4 +239,18 @@ public function indexAction(Request $request, LanguageHelper $languageHelper, Us
             ]
         );
     }
+
+    public function setPreferenceAction(Request $request, UserHelper $userHelper, UserModel $userModel): JsonResponse
+    {
+        $data  = json_decode($request->getContent(), true);
+        $key   = $data['preference'] ?? null;
+        $value = $data['value'] ?? null;
+
+        // Get current user
+        $user = $userHelper->getUser();
+
+        $userModel->setPreference($key, $value, $user);
+
+        return new JsonResponse(['success' => true]);
```

Comment:
I know that some old controllers respond with the success message but it's a duplicated information. That's what the 200 response code is for. Let's remove it and respond with empty JSON array if nothing else is needed.